### PR TITLE
[WIP][integration -- later?] python 2/3 compatibilities

### PR DESCRIPTION
**Pull request by fstagni**
A bunch of python 2/3 compatibilities. After these changes I could do `import DIRAC` within python 3.7.

A couple things:
- MySQL-python is not python 3 compatible, so I used https://pypi.org/project/mysqlclient/ which seems like the best option. Didn't really try it fully though.
- fts3-rest: I couldn't find a Python 3 version, but maybe simply it's not on pypi.

### CODE REVIEW

**Comment by chaen** [comment]
the py3 version of fts3-rest is on their roadmap

File: Core/DISET/private/BaseClient.py [hunk 1]
```diff
@@ -60,7 +64,7 @@ def __init__(self, serviceName, **kwargs):
       :param keepAliveLapse: Duration for keepAliveLapse (heartbeat like)
     """
 
-    if not isinstance(serviceName, basestring):
+    if not isinstance(serviceName, str):
```
Comment (chaen) [inline Core/DISET/private/BaseClient.py]: `basestring` disappear ?

File: Core/DISET/private/BaseClient.py [hunk 2]
```diff
@@ -400,7 +404,7 @@ def __checkThreadID(self):
     """
     if not self.__initStatus['OK']:
       return self.__initStatus
-    cThID = thread.get_ident()
+    cThID = threading.ident
```
Comment (chaen) [inline Core/DISET/private/BaseClient.py]:
```
In [13]: threading.ident
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-13-ab6e47a0c388> in <module>()
----> 1 threading.ident

AttributeError: 'module' object has no attribute 'ident'
```

File: Core/DISET/private/BaseClient.py (hunk 1 above)
Comment (fstagni) [reply Core/DISET/private/BaseClient.py]: yes!

File: Core/DISET/private/BaseClient.py (hunk 2 above)
Comment (fstagni) [reply Core/DISET/private/BaseClient.py]: fixed.

Comment (chaen) [reply Core/DISET/private/BaseClient.py]: threading does not have a get_ident call either :-/

**Comment by fstagni** [comment]
This PR was just a trial to see how far we are from a python 3 compatible version, and it's not meant for being merged anytime soon.
Most of the errors that I see in the unit tests are because of unicode, which as we know is something for which we need to pay attention from now on. The `DEncode` module is especially affected, but here we'll first need to see if we'll still need it.

I will leave this PR as it is for now, open for comments.

**Comment by fstagni** [comment]
BTW: I tried, in my local installation, to run some tests on python 3.7. Unsurprisingly, with results very similar to those for python 2.7 that you see in Travis.

File: ConfigurationSystem/Client/Utilities.py [hunk 3]
```diff
@@ -11,12 +11,16 @@
...
+from __future__ import absolute_import, print_function, unicode_literals
+
 __RCSID__ = "$Id$"
 
 import six
 import re
 import socket
-from urlparse import urlparse
+
+from builtins import str
```
Comment (fstagni) [inline ConfigurationSystem/Client/Utilities.py]: I rebased and this is the result... but I think this is wrong now... ?

**Comment by fstagni** [comment]
Closing now, it should be superseded.

**Pull request closed without merging**